packages/health/lib/src/health_factory.dart

```diff
@@ -125,6 +140,22 @@ class HealthFactory {
       }
     }
 
+    if (permissions != null) {
+      for (int i = 0; i < types.length; i++) {
+        final type = types[i];
+        final permission = permissions[i];
+        if ((type == HealthDataType.ELECTROCARDIOGRAM ||
+            type == HealthDataType.HIGH_HEART_RATE_EVENT ||
+            type == HealthDataType.LOW_HEART_RATE_EVENT ||
+            type == HealthDataType.IRREGULAR_HEART_RATE_EVENT ||
+            type == HealthDataType.WALKING_HEART_RATE) &&
+            permission != HealthDataAccess.READ) {
+          throw ArgumentError(
```

Comment:
why are we not throwing HealthException here?
will throwing ArgumentError crash the app?

Reply:
This part of code was taken from the plugin dev, but I will move to the `HealthException`. In fact there is not much difference, as both the Error and Exceptions while not caught are crashing the app.